modules/tinymce/src/core/main/ts/fmt/RemoveFormat.ts

```diff
@@ -533,30 +544,33 @@ const remove = function (ed: Editor, name: string, vars?: FormatVars, node?: Nod
     }
 
     // Remove items between start/end
-    RangeWalk.walk(dom, expandedRng, function (nodes) {
-      each(nodes, function (node) {
+    RangeWalk.walk(dom, expandedRng, (nodes) => {
+      Arr.each(nodes, (node) => {
         process(node);
 
-        // Remove parent span if it only contains text-decoration: underline, yet a parent node is also underlined.
-        if (NodeType.isElement(node) && ed.dom.getStyle(node, 'text-decoration') === 'underline' &&
-          node.parentNode && FormatUtils.getTextDecoration(dom, node.parentNode) === 'underline') {
-          removeFormat(ed, {
-            deep: false,
-            exact: true,
-            inline: 'span',
-            styles: {
-              textDecoration: 'underline'
-            }
-          }, null, node);
-        }
+        // Remove parent span if it only contains text-decoration, yet a parent node also has the same text decoration.
+        const textDecorations = [ 'underline', 'line-through', 'overline' ];
```

Comment:
If it's easy, could we add a test case for this too please? I know it's tangential, but would be nice.

Reply:
Added a test. Thought I should note that It appears a couple of tests fail when the text-decoration logic is altered/removed, namely `underline colors 6` and `underline colors 7` in FormatterApplyTest.ts so there should be enough coverage for this logic.